xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -43,6 +42,23 @@ impl StoredGroup {
     }
 }
 
+impl EncryptedMessageStore {
+    /// Updates group membership state
+    pub fn update_group_membership<GroupId: AsRef<ID>>(
+        &self,
+        conn: &mut DbConnection,
+        id: GroupId,
+        state: GroupMembershipState,
+    ) -> Result<StoredGroup, StorageError> {
+        use super::schema::groups::dsl;
+
+        diesel::update(dsl::groups.find(id.as_ref()))
+            .set(dsl::membership_state.eq(state))
+            .get_result::<StoredGroup>(conn)
```

Comment:
Do we actually need to return the value here? I've been working on the state machine, and it's all very much "pull from the DB when you need to know something" so there's no chance of out-of-date state in memory when doing concurrent work.

Reply:
Yeah don't think so, I just tried to mirror what was in `xmtp` as far as these updates look like (lacking some in-code context on state machine overall atm, but hope to get up to speed on that this week with a ticket) -- I'm all for less complexity wherever possible